src/emmy/abstract/function.cljc

```diff
@@ -262,15 +277,50 @@
            (g/+ tangent (g/* (literal-apply partial primal-s)
                              dx))))))))
 
+(defn- reverse-mode-fold
+  "Takes
+
+    - a literal function `f`
+    - a structure `primal-s` of the primal components of the args to `f` (with
+      respect to `tag`)
+    - the `tag` of the innermost active derivative call
+
+  And returns a folding function (designed for use
+  with [[emmy.structure/fold-chain]]) that assembles all partial derivatives of
+  `f` into a new [[emmy.tape/TapeCell]]."
+  [f primal-s tag]
+  (fn
+    ([] [])
+    ([partials]
+     (tape/make tag (apply f primal-s) partials))
+    ([partials [entry path _]]
+     (if (and (tape/tape? entry) (= tag (tape/tape-tag entry)))
+       (let [partial (literal-partial f path)]
+         (conj partials [entry (literal-apply partial primal-s)]))
+       partials))))
```

Comment:
this would happen in the case of nested `gradient` calls to a literal function, I just don't have those in the tests yet. In that case, the innermost tag wins and any tag with a different tape is treated as a scalar (i.e. partial derivative == 0, so we never add it to the map)

Reply:
I want to share tests between forward mode and reverse mode... when I do that this should get hit.